modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQJMSContext.java

```diff
@@ -0,0 +1,729 @@
+            logger.debug("Created Temporary Queue");
+            return temporaryQueue;
+        } catch (JMSException e) {
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public TemporaryTopic createTemporaryTopic() {
+        try {
+            TemporaryTopic temporaryTopic = getSession().createTemporaryTopic();
+            logger.debug("Created Temporary Topic");
+            return temporaryTopic;
+        } catch (JMSException e) {
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     * TODO: Should throw InvalidDestinationRuntimeException if invalid name.
+     *
+     * @param name name used to identify this subscription
+     */
+    @Override
+    public void unsubscribe(String name) {
+        try {
+            getSession().unsubscribe(name);
+        } catch (JMSException e) {
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void acknowledge() {
+        try {
+            getSession().acknowledge();
+        } catch (IllegalStateException e) {
+            throw new IllegalStateRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * Method to retrieve session.
+     * Creates session if not already created.
+     *
+     * @return session currently applicable session
+     */
+    private AMQSession getSession() {
+        if (session == null) {
+            try {
+                session = (AMQSession) connection.createSession(getSessionMode());
+            } catch (JMSException e) {
+                throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+            }
+        }
+        return session;
+    }
+
+    /**
+     * Method to retrieve the single message producer of the JMSContext.
+     * Creates message producer if not already created.
+     *
+     * @return MessageProducer the shared MessageProducer
+     */
+    private MessageProducer getSharedMessageProducer() {
+        if (sharedMessageProducer == null) {
+            try {
+                sharedMessageProducer = session.createProducer(null);
+            } catch (java.lang.IllegalStateException e) {
+                throw new IllegalStateRuntimeException(e.getMessage(), null, e);
+            } catch (JMSException e) {
+                throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+            }
+        }
+        return sharedMessageProducer;
+    }
+
+    /**
+     * According to the spec: connection needs to be started automatically when a consumer is
+     * created if autoStart is set to true.
+     */
+    private void startConnectionIfRequired() {
+        if (getAutoStart()) {
+            try {
+                if (!connection._started) {
```

Comment:
The _started variable is declared in the AMQConnection class (based on qpid code) and used across multiple classes to identify if the connection has been started. Needs to be fixed in all places.